sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -0,0 +1,760 @@
+     * Gets the list of the acquired phone numbers.
+     *
+     * @param locale A language-locale pairing which will be used to localise the names of countries.
+     * @return A {@link PagedFlux} of {@link AcquiredPhoneNumber} instances representing acquired telephone numbers.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<AcquiredPhoneNumber> listAllPhoneNumbers(String locale) {
+        return listAllPhoneNumbers(locale, null);
+    }
+
+    PagedFlux<AcquiredPhoneNumber> listAllPhoneNumbers(String locale, Context context) {
+        try {
+            if (context == null) {
+                return phoneNumberAdministrations.getAllPhoneNumbersAsync(locale, null, null);
+            } else {
+                return phoneNumberAdministrations.getAllPhoneNumbersAsync(locale, null, null, context);
+            }
+        } catch (RuntimeException ex) {
+            return pagedFluxError(logger, ex);
+        }
+    }
+
+    /**
+     * Gets a list of the supported area codes.
+     *
+     * @param locationType The type of location information required by the plan.
+     * @param countryCode The ISO 3166-2 country code.
+     * @param phonePlanId The plan id from which to search area codes.
+     * @param locationOptions A {@link List} of {@link LocationOptionsQuery} for querying the area codes.
+     * @return A {@link Mono} containing a {@link AreaCodes} representing area codes.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<AreaCodes> getAllAreaCodes(
+        String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions) {
+        return getAllAreaCodesWithResponse(locationType, countryCode, phonePlanId, locationOptions)
+            .flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Gets a list of the supported area codes.
+     *
+     * @param locationType The type of location information required by the plan.
+     * @param countryCode The ISO 3166-2 country code.
+     * @param phonePlanId The plan id from which to search area codes.
+     * @param locationOptions A {@link List} of {@link LocationOptionsQuery} for querying the area codes.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue()} value returns
+     * a {@link AreaCodes} representing area codes.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<AreaCodes>> getAllAreaCodesWithResponse(
+        String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions) {
+        return getAllAreaCodesWithResponse(locationType, countryCode, phonePlanId, locationOptions, null);
+    }
+
+    Mono<Response<AreaCodes>> getAllAreaCodesWithResponse(
+        String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions,
+        Context context) {
+        Objects.requireNonNull(locationType, "'locationType' cannot be null.");
```

Comment:
Methods that return reactive publishers should return error through Mono.error or Flux.error instead of throwing exceptions.